CMakeLists.txt

```diff
@@ -511,42 +506,55 @@ message(STATUS "  Slicer_VTK_VERSION_MAJOR is ${Slicer_VTK_VERSION_MAJOR}")
 #-----------------------------------------------------------------------------
 # Slicer_VTK_COMPONENTS
 #-----------------------------------------------------------------------------
+if (Slicer_VTK_VERSION_MAJOR VERSION_LESS "9")
+  set(VTK_COMPONENT_PREFIX "vtk")
+  set(VTK_TARGET_PREFIX "vtk")
+else()
+  set(VTK_COMPONENT_PREFIX "")
+  set(VTK_TARGET_PREFIX "VTK::")
+endif()
 set(Slicer_VTK_COMPONENTS
-  vtkFiltersExtraction
-  vtkFiltersFlowPaths
-  vtkFiltersGeometry
-  vtkGUISupportQtOpenGL
-  vtkGUISupportQtSQL
-  vtkIOExport
-  vtkIOImage
-  vtkIOLegacy
-  vtkIOPLY
-  vtkIOXML
-  vtkImagingMath
-  vtkImagingMorphological
-  vtkImagingStatistics
-  vtkImagingStencil
-  vtkInteractionImage
-  vtkRenderingContext${Slicer_VTK_RENDERING_BACKEND}
-  vtkRenderingQt
-  vtkRenderingVolume${Slicer_VTK_RENDERING_BACKEND}
-  vtkTestingRendering
-  vtkViewsQt
-  vtkzlib
+  ${VTK_COMPONENT_PREFIX}FiltersExtraction
+  ${VTK_COMPONENT_PREFIX}FiltersFlowPaths
+  ${VTK_COMPONENT_PREFIX}FiltersGeometry
+  ${VTK_COMPONENT_PREFIX}FiltersParallel
+  ${VTK_COMPONENT_PREFIX}GUISupportQtSQL
+  ${VTK_COMPONENT_PREFIX}IOExport
+  ${VTK_COMPONENT_PREFIX}IOImage
+  ${VTK_COMPONENT_PREFIX}IOLegacy
+  ${VTK_COMPONENT_PREFIX}IOPLY
+  ${VTK_COMPONENT_PREFIX}IOXML
+  ${VTK_COMPONENT_PREFIX}ImagingMath
+  ${VTK_COMPONENT_PREFIX}ImagingMorphological
+  ${VTK_COMPONENT_PREFIX}ImagingStatistics
+  ${VTK_COMPONENT_PREFIX}ImagingStencil
+  ${VTK_COMPONENT_PREFIX}InteractionImage
+  ${VTK_COMPONENT_PREFIX}RenderingContext${Slicer_VTK_RENDERING_BACKEND}
```

Comment:
VTK 8 didn't support a non-`OpenGL2` backend either. This could probably just be `OpenGL2` rather than being a variable these days (but is likely a change for another PR since I suspect even more code could be simplified).